dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ReturnTypeChangedProblem.java

```diff
@@ -24,15 +26,32 @@
  * actualTypeName}).
  */
 class ReturnTypeChangedProblem extends LinkageProblem {
+  private String actualType;
+
   ReturnTypeChangedProblem(
       ClassFile sourceClass,
       @Nullable ClassFile targetClass,
       MethodSymbol expectedMethodSymbol,
       String actualType) {
     super(
-        "is not found. The expected return type does not match the actual type " + actualType,
+        "is expected to return "
+            + Utility.methodSignatureReturnType(expectedMethodSymbol.getDescriptor())
+            + " but the actual type is "
+            + actualType,
         sourceClass,
         expectedMethodSymbol,
         targetClass);
+    this.actualType = actualType;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(super.hashCode(), actualType);
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    return super.equals(other) // this checks the class equality
+        && Objects.equals(actualType, ((ReturnTypeChangedProblem) other).actualType);
```

Comment:
These overrides have been missing and allowed `testFindLinkageProblems_referenceToJava11Method` to ignore the `actualType` field.

Reply:
Good catch. The equals method in the supertype should perhaps be testing for class equality.

Reply:
Added test with ReturnTypeChangedProblem.